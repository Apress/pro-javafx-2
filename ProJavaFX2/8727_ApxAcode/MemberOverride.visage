/*
 * @test
 * @run
 */

class Point {
  var x: Number;
  var y: Number;
}

var p = Point {
  override var x on replace {
    println("x is now {x}");
  }
  override function toString(): String {
    "Point({x}, {y})"
  }
  x: 3.0
  y: 4.0
};

println(p);
