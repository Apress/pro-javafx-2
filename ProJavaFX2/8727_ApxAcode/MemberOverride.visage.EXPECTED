x is now 3.0
Point(3.0, 4.0)
